chia/wallet/vc_wallet/vc_wallet.py

```diff
@@ -0,0 +1,460 @@
+        """
+        wallet_node = self.wallet_state_manager.wallet_node
+        coin_states: Optional[List[CoinState]] = await wallet_node.get_coin_state([coin.parent_coin_info], peer=peer)
+        if coin_states is None:
+            self.log.error(f"Cannot find parent coin of the verified credential coin: {coin.name().hex()}")
+            return
+        parent_coin_state = coin_states[0]
+        cs = await fetch_coin_spend_for_coin_state(parent_coin_state, peer)
+        if cs is None:
+            self.log.error(f"Cannot get verified credential coin: {coin.name().hex()} puzzle and solution")
+            return
+        vc = VerifiedCredential.get_next_from_coin_spend(cs)
+        vc_record: VCRecord = VCRecord(vc, height)
+        await self.store.add_or_replace_vc_record(vc_record)
+
+    async def remove_coin(self, coin: Coin, height: uint32) -> None:
+        """
+        remove the VC if it is transferred to another key
+        :param coin:
+        :param height:
+        :return:
+        """
+        vc_record: Optional[VCRecord] = await self.store.get_vc_record_by_coin_id(coin.name())
+        if vc_record is not None:
+            await self.store.delete_vc_record(vc_record.vc.launcher_id)
+
+    async def get_vc_record_for_launcher_id(self, launcher_id: bytes32) -> VCRecord:
+        """
+        Go into the store and get the VC Record representing the latest representation of the VC we have on chain.
+        """
+        vc_record = await self.store.get_vc_record(launcher_id)
+        if vc_record is None:
+            raise ValueError(f"Verified credential {launcher_id.hex()} doesn't exist.")
+        return vc_record
+
+    async def launch_new_vc(
+        self,
+        provider_did: bytes32,
+        inner_puzzle_hash: Optional[bytes32] = None,
+        fee: uint64 = uint64(0),
+    ) -> Tuple[VCRecord, List[TransactionRecord]]:
+        """
+        Given the DID ID of a proof provider, mint a brand new VC with an empty slot for proofs.
+
+        Returns the tx records associated with the transaction as well as the expected unconfirmed VCRecord.
+        """
+        # Check if we own the DID
+        found_did = False
+        for _, wallet in self.wallet_state_manager.wallets.items():
+            if wallet.type() == WalletType.DECENTRALIZED_ID:
+                assert isinstance(wallet, DIDWallet)
+                if bytes32.fromhex(wallet.get_my_DID()) == provider_did:
+                    found_did = True
+                    break
+        if not found_did:
+            raise ValueError(f"You don't own the DID {provider_did.hex()}")
+        # Mint VC
+        coins = await self.standard_wallet.select_coins(uint64(2 + fee), min_coin_amount=uint64(2 + fee))
+        if len(coins) == 0:
+            raise ValueError("Cannot find a coin to mint the verified credential.")
+        if inner_puzzle_hash is None:
+            inner_puzzle_hash = await self.standard_wallet.get_puzzle_hash(new=False)
```

Comment:
does it always need to use the old puzzle hash?

Reply:
I don't think there's much of an argument for privacy here, it's very clear that what you're doing is launching a VC and very likely to yourself.  Not generating a new puzzle hash here just keeps the puzzle DB a little slimmer.